### ISSUE TITLE
Support Top-K data structure and commands

### ISSUE THREAD

**Issue by PragmaTwice** [apache/kvrocks#2424]
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/kvrocks/issues) and found no similar issues.

### Motivation

Redis Stack already supports this probabilistic data structure, and we can support it too.

https://redis.io/docs/latest/develop/data-types/probabilistic/top-k/
https://redis.io/docs/latest/commands/?group=topk

### Solution

Please share your design proposal through issue/discussion before submitting a PR.

### Are you willing to submit a PR?

- [ ] I'm willing to submit a PR!

**Comment by zweix123** [comment]
I hope to complete this issue. Can you assign it to me.

**Comment by git-hulk** [comment]
@zweix123 Assigned, thank you! Could you please propose the design before starting implementation? So that we can discuss the design in the early stage.

**Comment by zweix123** [comment]
> @zweix123 Assigned, thank you! Could you please propose the design before starting implementation? So that we can discuss the design in the early stage.

Yes, I have such thoughts. Please give me some time, it won't be too long, about a day.

**Comment by git-hulk** [comment]
@zweix123 Thank you! There's no rush, it's good to propose the design while you get time and are ready to review.

**Comment by zweix123** [comment]
A considerable amount of design has been completed, but there are some difficulties that will be discussed here: [Kvrocks-Discussions-Difficulties in designing documents for Top-K data structures and commands #2449](https://github.com/apache/kvrocks/discussions/2449).
